overlay.d/05rhcos/usr/libexec/rhcos-fail-boot-for-older-luks-syntax

```diff
@@ -0,0 +1,23 @@
+#!/bin/bash
+set -euo pipefail
+
+# This script tries to look for a `/etc/clevis.json` reference 
+# (used to enable LUKS in RHCOS 4.6 or earlier versions) in an 
+# Ignition config. If it exists and the version of RHCOS is 4.7, 
+# then the script ends with exit 1.
+
+ignition_cfg="/run/ignition.json"
+wanted_path="/etc/clevis.json"
+tmpconfig=${outputdir}/tmpconfig.json
+
+source /etc/os-release
+if [ "$VERSION_ID" == "4.7" ]; then
+    # select the `/etc/clevis.json` entry from a given Ignition config
+    (jq ".storage.files[] | select(.path==\"${wanted_path}\")") \
+        < "${ign_config}" > ${tmpconfig}.fragment
+
+    if [ -s ${tmpconfig}.fragment ]; then
+        echo "Failed to boot due to an older syntax used to configure LUKS"
```

Comment:
I think we should get a PR into the OpenShift docs to describe this and change this to hyperlink to that.

Reply:
Should probably be added to https://github.com/openshift/openshift-docs/pull/27661.